ivadomed/transforms.py

```diff
@@ -1139,6 +1141,60 @@ def __call__(self, sample, metadata=None):
             return sample, metadata
 
 
+class Denoise(ImedTransform):
+    """Denoise an image using non-local means adaptive denoising from P. Coupe et al. as implemented
+    in dipy
+
+    Args:
+        patch_radius (int): patch size equals 2 x patch radius + 1
+        block_radius (int): block size equals 2 x block radius + 1
+    """
+
+    def __init__(self, patch_radius=1, block_radius=5):
+        self.patch_radius = patch_radius
+        self.block_radius = block_radius
+
+    def denoise_nlmeans(self, sample):
+        """In-class function for denoising with nlmeans as implemented in spinalcordtoolbox.math"""
+        sample = np.asarray(sample)
+
+        block_radius_max = min(sample.shape) - 1
+        block_radius = block_radius_max if self.block_radius > block_radius_max else self.block_radius
+
+        sigma = estimate_sigma(sample)
+        denoised = nlmeans(sample, sigma, patch_radius=self.patch_radius, block_radius=block_radius)
+
+        return denoised
+
+    @multichannel_capable
+    @two_dim_compatible
+    def undo_transform(self, sample, metadata=None):
+        """Placeholder for consistency with preprocessing steps"""
+        pass
```

Comment:
Minor nitpick could be ignored as well, since it's not possible to undo this transform, wouldn't it be better if we just skipped this?